File: android/app/src/main/java/com/appxstudios/festivalconnection/ui/theme/FestivalTheme.kt
package com.appxstudios.festivalconnection.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.LinearGradientShader
import androidx.compose.ui.graphics.ShaderBrush
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.Text
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.drawWithCache
import androidx.compose.ui.graphics.drawscope.Fill
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.foundation.layout.size
import androidx.compose.material3.Icon
import androidx.compose.ui.draw.drawWithContent
import androidx.compose.ui.graphics.BlendMode
import androidx.compose.ui.graphics.graphicsLayer
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

// Gradient stops (4-stop purple-fuchsia-pink-tangerine — matches iOS exactly)
val GradientPurple = Color(0xFF7B2FBE)
val GradientFuchsia = Color(0xFFC026D3)
val GradientPink = Color(0xFFEC4899)
val GradientTangerine = Color(0xFFF59E0B)

// Backward-compat aliases
val GradientMagenta = GradientFuchsia
val GradientCoral = GradientPink
val GradientOrange = GradientTangerine
val GradientAmber = GradientTangerine

// Accent
val AccentPink = GradientPink

// Backgrounds & Surfaces
val BackgroundBlack = Color(0xFF000000)
val SurfaceDark = Color(0xFF1C1C1E)
val SurfaceMedium = Color(0xFF2C2C2E)

// Text
val TextPrimary = Color.White
val TextSecondary = Color(0xFF999999)
val TextMuted = Color(0xFF666666)

// Status
val PresenceGreen = Color(0xFF32D74B)
val ErrorRed = Color.Red

// Payment
val PaymentBorder = Color(0xFF7A3A00)

// Icon Colors (kept for backward compat)
val IconOrange = GradientOrange
val IconBlue = Color(0xFF409CFF)
val IconPurple = GradientPurple
val IconGreen = Color(0xFF32D74B)
val IconRed = Color(0xFFFF453A)
val IconGold = Color(0xFFFFCC00)
val IconGray = Color(0xFF8C8C8C)

val MainGradientColors = listOf(GradientPurple, GradientFuchsia, GradientPink, GradientTangerine)

fun mainGradientBrush(): Brush = Brush.linearGradient(colors = MainGradientColors)

fun mainGradientVerticalBrush(): Brush = Brush.verticalGradient(colors = MainGradientColors)

fun mainGradientDiagonalBrush(): Brush = Brush.linearGradient(
    colors = MainGradientColors,
    start = Offset(0f, 0f),
    end = Offset(1000f, 1000f)
)

private val FestivalColorScheme = darkColorScheme(
    background = BackgroundBlack,
    surface = SurfaceDark,
    primary = AccentPink,
    onPrimary = Color.White,
    secondary = GradientPurple,
    onBackground = Color.White,
    onSurface = Color.White,
)

@Composable
fun FestivalConnectionTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = FestivalColorScheme,
        content = content
    )
}

@Composable
fun GradientText(
    text: String,
    modifier: Modifier = Modifier,
    fontSize: androidx.compose.ui.unit.TextUnit = 16.sp,
    fontWeight: FontWeight = FontWeight.Normal,
    textAlign: TextAlign? = null,
    letterSpacing: androidx.compose.ui.unit.TextUnit = androidx.compose.ui.unit.TextUnit.Unspecified
) {
    Text(
        text = text,
        modifier = modifier,
        fontSize = fontSize,
        fontWeight = fontWeight,
        textAlign = textAlign,
        letterSpacing = letterSpacing,
        style = TextStyle(
            brush = mainGradientBrush()
        )
    )
}

@Composable
fun GradientIcon(
    imageVector: ImageVector,
    contentDescription: String? = null,
    size: Dp = 24.dp,
    modifier: Modifier = Modifier
) {
    Icon(
        imageVector = imageVector,
        contentDescription = contentDescription,
        modifier = modifier
            .size(size)
            .graphicsLayer(alpha = 0.99f)
            .drawWithContent {
                drawContent()
                drawRect(
                    brush = Brush.linearGradient(MainGradientColors),
                    blendMode = BlendMode.SrcAtop
                )
            },
        tint = Color.White
    )
}

File: ios/FestivalConnection/UI/Theme/FestivalTheme.swift
import SwiftUI

// MARK: - Festival Connection Theme
// Single source of truth for every color and gradient value in the entire app.
// Do not define any color value outside this file.
// Do not hardcode any hex color value outside this file.

enum FestivalTheme {
    // MARK: - Gradient Stops (4-stop purple-fuchsia-pink-tangerine)
    static let gradientPurple    = Color(red: 0.482, green: 0.184, blue: 0.745) // #7B2FBE
    static let gradientFuchsia   = Color(red: 0.752, green: 0.149, blue: 0.824) // #C026D3
    static let gradientPink      = Color(red: 0.925, green: 0.286, blue: 0.600) // #EC4899
    static let gradientTangerine = Color(red: 0.961, green: 0.620, blue: 0.043) // #F59E0B

    // Backward-compat aliases for existing code referencing old names
    static let gradientMagenta = gradientFuchsia
    static let gradientCoral   = gradientPink
    static let gradientOrange  = gradientTangerine
    static let gradientAmber   = gradientTangerine

    // MARK: - Accent / Interactive
    static let accentPink = gradientPink  // buttons, FAB, send, active tab, links

    // MARK: - Backgrounds
    static let backgroundBlack = Color(red: 0.0, green: 0.0, blue: 0.0)      // Pure #000000
    static let surfaceDark     = Color(red: 0.110, green: 0.110, blue: 0.118) // ~#1C1C1E
    static let surfaceMedium   = Color(red: 0.173, green: 0.173, blue: 0.180) // ~#2C2C2E

    // MARK: - Text
    static let textPrimary   = Color.white
    static let textSecondary = Color(white: 0.60)
    static let textMuted     = Color(white: 0.40)

    // MARK: - Status
    static let presenceGreen = Color(red: 0.196, green: 0.843, blue: 0.294)
    static let errorRed      = Color.red
    static let bitcoinOrange = Color(red: 0.478, green: 0.224, blue: 0.0) // ~#7A3A00

    // Backward-compat alias
    static let paymentBorder = bitcoinOrange

    // MARK: - Settings Icon Colors (individual tones)
    static let iconOrange = Color(red: 1.0, green: 0.58, blue: 0.0)
    static let iconBlue   = Color(red: 0.0, green: 0.48, blue: 1.0)
    static let iconPurple = Color(red: 0.58, green: 0.24, blue: 0.83)
    static let iconGreen  = Color(red: 0.20, green: 0.78, blue: 0.35)
    static let iconPink   = Color(red: 0.93, green: 0.29, blue: 0.60)
    static let iconRed    = Color(red: 1.0, green: 0.27, blue: 0.23)
    static let iconGold   = Color(red: 1.0, green: 0.84, blue: 0.0)
    static let iconGray   = Color(white: 0.55)

    // MARK: - Gradient Expressions

    static let mainGradientColors: [Color] = [
        gradientPurple, gradientFuchsia, gradientPink, gradientTangerine
    ]

    static let mainGradient = LinearGradient(
        colors: mainGradientColors,
        startPoint: .leading,
        endPoint: .trailing
    )

    static let mainGradientVertical = LinearGradient(
        colors: mainGradientColors,
        startPoint: .top,
        endPoint: .bottom
    )

    static let mainGradientDiagonal = LinearGradient(
        colors: mainGradientColors,
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )
}
